pages/datasets/_datasetId.vue

```diff
@@ -294,24 +295,43 @@ const getThumbnailData = async (datasetDoi, datasetId, datasetVersion, datasetFa
             species = speciesArray[0].children[0].label.toLowerCase()
         }
         let taxo = Uberons.species['rat']
-        if (species && (species in Uberons.species))
+
+        if (species && species in Uberons.species)
           taxo = Uberons.species[species]
-          
-        scicrunchData.organs.forEach(organ => {
-          let organData = {
-            taxo,
-            uberonid: organ.curie,
-            organ: organ.name,
-            id: datasetId,
-            version: datasetVersion
+
+        // Check if flatmap has the anatomy for this species. This is done by asking the flatmap knowledge base
+        // if a flatmap of (species) has (anatomy)
+        let foundAnatomy = []
+        if (scicrunchData.organs[0]) { // Check if dataset has organ annotation
+          const anatomy = scicrunchData.organs.map(organ => organ.curie)
+          let data = await flatmaps.anatomyQuery(taxo, anatomy)
+          let anatomyResponse = data.data.values
```

Comment:
Does the response contain data even in failure? Better check or the page would not have the SciCrunch and Biolucida data...

Reply:
yeah good question I will have a look at this